src/aks-preview/azext_aks_preview/custom.py

```diff
@@ -1250,28 +1252,34 @@ def aks_create(cmd,     # pylint: disable=too-many-locals,too-many-statements,to
             )
 
     addon_profiles = _handle_addons_args(
-        cmd,
-        enable_addons,
-        subscription_id,
-        resource_group_name,
-        {},
-        workspace_resource_id,
-        appgw_name,
-        appgw_subnet_prefix,
-        appgw_subnet_cidr,
-        appgw_id,
-        appgw_subnet_id,
-        appgw_watch_namespace,
-        enable_sgxquotehelper,
-        aci_subnet_name,
-        vnet_subnet_id,
-        enable_secret_rotation
+        cmd = cmd,
+        addons_str = enable_addons,
+        subscription_id = subscription_id,
+        resource_group_name = resource_group_name,
+        addon_profiles = {},
+        workspace_resource_id = workspace_resource_id,
+        enable_msi_auth_for_monitoring = enable_msi_auth_for_monitoring,
+        appgw_name = appgw_name,
+        appgw_subnet_prefix = appgw_subnet_prefix,
+        appgw_subnet_cidr = appgw_subnet_cidr,
+        appgw_id = appgw_id,
+        appgw_subnet_id = appgw_subnet_id,
+        appgw_watch_namespace = appgw_watch_namespace,
+        enable_sgxquotehelper = enable_sgxquotehelper,
+        aci_subnet_name = aci_subnet_name,
+        vnet_subnet_id = vnet_subnet_id,
+        enable_secret_rotation = enable_secret_rotation,        
     )
     monitoring = False
     if CONST_MONITORING_ADDON_NAME in addon_profiles:
         monitoring = True
-        _ensure_container_insights_for_monitoring(
-            cmd, addon_profiles[CONST_MONITORING_ADDON_NAME])
+        if enable_msi_auth_for_monitoring and not enable_managed_identity:
+            raise CLIError("--enable-msi-auth-for-monitoring can not be used on clusters with service principal auth.")
```

Comment:
Could you please use the specific error type `ArgumentUsageError` instead of `CLIError`